### DISCUSSION TITLE
Use of Windows.Services.Store WinRT API for desktop EXE applications in Microsoft Store

### DISCUSSION THREAD

**Discussion by geniuss99** [microsoft/CsWinRT#1793] [Q&A] [answered]
I would like to use the `Windows.Services.Store` **WinRT API** in one of my desktop non-packaged products which is already in **Microsoft Store** to parse the campaign ID upon installation. I used the following code as an example:

https://blogs.windows.com/windowsdeveloper/2019/04/30/calling-windows-10-apis-from-a-desktop-application-just-got-easier/
https://learn.microsoft.com/en-us/windows/apps/publish/create-a-custom-app-promotion-campaign

According to the documentation `Windows.Services.Store` **WinRT API** requires a process token to have a package identity. As my program is not packaged (no **MSIX** container) it does not have a package identity. To circumvent this limitation I used packaging with external location as described in MSDN:
https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/grant-identity-to-nonpackaged-apps

Now to use the `Windows.Services.Store` **WinRT API** it looks like I need to use the correct package identity in the code. For **MSIX** type applications this information is located in the Product management->Product identity page of **Microsoft Partner Center**. There is no such information for **EXE** type applications though.

Is there any product identity for EXE type applications in the **Microsoft Store** which can be used for `Windows.Services.Store` **WinRT API**? Or does anyone have solutions to the current problem?

**Comment by dongle-the-gadget** [comment, marked as answer]
It won’t work. The API requires that the full package be available in the Microsoft Store.
